pkg/resourceinterpreter/default/native/status_type.go

```diff
@@ -0,0 +1,26 @@
+/*
+Copyright 2024 The Karmada Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package native
+
+import appsv1 "k8s.io/api/apps/v1"
+
+// WrappedDeploymentStatus is a wrapper for appsv1.DeploymentStatus.
+type WrappedDeploymentStatus struct {
```

Comment:
This method means that we need to add a status wrapper for each supported resource in the future. For resources implemented in a customized manner, for example, a crd, it may not be easy to handle.

I was wondering if we could add these required fields to the API definition of work.

Reply:
> This method means that we need to add a status wrapper for each supported resource in the future.

I guess that's the idea.
Given we want to figure out whether the resource has been `synced` or not, we need to get the `.metadata.generation`, but no place to hold it in the native `appsv1.DeploymentStatus`.

Reply:
> For resources implemented in a customized manner, for example, a crd, it may not be easy to handle.

That's a good point!
Indeed, this approach brings a limitation for those who use customized interpreters. That is:
`If people customized InterpretStatus, then he must customize AggregateStatus as well`, because they both relying a same structure.

Reply:
> I was wondering if we could add these required fields to the API definition of work.

It's not easy to do that, as the [ReflectStatus](https://github.com/karmada-io/karmada/blob/3232c52d57b331d7120eeaac9386b848197475df/pkg/resourceinterpreter/interpreter.go#L62) interface not referencing the Work object(and it shouldn't).

Reply:
> It's not easy to do that, as the [ReflectStatus](https://github.com/karmada-io/karmada/blob/3232c52d57b331d7120eeaac9386b848197475df/pkg/resourceinterpreter/interpreter.go#L62) interface not referencing the Work object(and it shouldn't).

You are right. Modifying the work API is not necessarily a good idea.